src/pal/src/exception/signal.cpp

```diff
@@ -554,6 +555,23 @@ static void common_signal_handler(PEXCEPTION_POINTERS pointers, int code,
         ASSERT("sigprocmask failed; error is %d (%s)\n",errno, strerror(errno));
     } 
 
+#if defined(_AMD64_)
+    if (pointers->ExceptionRecord->ExceptionCode == EXCEPTION_INT_DIVIDE_BY_ZERO)
+    {
+        // It is possible that an overflow may also have been mapped to a divide-by-zero
+        // exception. This happens when we try to divide the maximum negative value of a
+        // signed integer with -1. 
+        //
+        // Thus, we will attempt to decode the instruction @ RIP to determine if that
+        // is the case using the faulting context.
+        if (IsDivByZeroAnIntegerOverflow(pointers->ContextRecord, pointers->ExceptionRecord))
```

Comment:
Ah, I understand. But again, the worst thing that can happen is that we exit the process due to the fact that we calculate a wrong address and an attempt to dereference it would lead to SIGSEGV. We tear down the process when div error happens out of our managed code anyways, so there seems to be no additional danger.
As for possibly missing some instruction encodings, I would like to write a verifier that checks all possible IDIV encodings using this code at some point to get full confidence. It should not be that difficult to have code that would generate the instructions based on the Intel architecture manual one by one, setting registers in the context so that we can verify the resulting address / value.